src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -260,9 +304,10 @@ private async Task<T> GetConfigurationNonBlockingAsync(CancellationToken cancel)
                 {
                     TelemetryClient.IncrementConfigurationRefreshRequestCounter(
                         MetadataAddress,
-                        TelemetryConstants.Protocols.Automatic);
+                        TelemetryConstants.Protocols.Automatic,
+                        TelemetryConstants.Protocols.ConfigurationSourceUnknown);
 
-                    _ = Task.Run(UpdateCurrentConfiguration, CancellationToken.None);
+                    _ = Task.Run(UpdateCurrentConfigurationAsync, CancellationToken.None);
```

Comment:
Should we pass a cancellation token here now since we are calling the event handlers in UpdateCurrentConfigurationAsync which can hang? Also propagate the cancellation token to the handlers?